src/aics-image-viewer/components/Toolbar/index.tsx

```diff
@@ -41,116 +42,214 @@ interface ToolbarProps {
-  const turntableToggleTitle = autorotate ? "Turn off turntable" : "Turn on turntable";
+  scrollX = (amount: number): number => (this.barRef.current!.scrollLeft += amount);
 
+  // Translate vertical scrolling into horizontal scrolling
+  wheelHandler: React.WheelEventHandler<HTMLDivElement> = (e) => {
+    e.preventDefault();
+    if (e.deltaY === 0) {
+      return;
+    }
+    this.scrollX(e.deltaY);
+  };
+
+  // Scroll buttons are only visible when toolbar can be scrolled in that direction.
+  // This may change on either scroll or resize.
+  checkScrollBtnVisible = (): void => {
+    const barEl = this.barRef.current;
+    if (!barEl) {
+      return;
+    }
+    const scrollBtnLeft = barEl.scrollLeft > 0;
+    const scrollBtnRight = barEl.scrollLeft < barEl.scrollWidth - barEl.clientWidth;
+    if (scrollBtnLeft !== this.state.scrollBtnLeft || scrollBtnRight !== this.state.scrollBtnRight) {
+      this.setState({ scrollBtnLeft, scrollBtnRight });
+    }
+  };
+
+  toggleAxis = (): void => this.props.changeAxisShowing(!this.props.showAxes);
+  toggleBoundingBox = (): void => this.props.changeBoundingBoxShowing(!this.props.showBoundingBox);
   // TODO remove ant-btn-icon-only hack when upgrading antd
-  const classForToggleBtn = (active: boolean): string =>
-    "ant-btn-icon-only btn-borderless" + (active ? " btn-active" : "");
-
-  return (
-    <div className="viewer-toolbar">
-      <span className="viewer-toolbar-center">
-        {renderGroup1 && (
-          <span className="viewer-toolbar-group">
-            {renderConfig.viewModeRadioButtons && (
-              <ViewModeRadioButtons mode={props.mode} onViewModeChange={props.onViewModeChange} />
+  classForToggleBtn = (active: boolean): string => "ant-btn-icon-only btn-borderless" + (active ? " btn-active" : "");
+
+  render(): React.ReactElement {
+    const { props } = this;
+    const { renderConfig, showAxes, showBoundingBox, autorotate } = props;
+    const { scrollMode, scrollBtnLeft, scrollBtnRight } = this.state;
+    const twoDMode = props.mode !== ViewMode.threeD;
+
+    const renderGroup1 =
+      renderConfig.viewModeRadioButtons || renderConfig.resetCameraButton || renderConfig.autoRotateButton;
+    const renderGroup4 = renderConfig.showAxesButton || renderConfig.showBoundingBoxButton;
+
+    const axesToggleTitle = showAxes ? "Hide axes" : "Show axes";
+    const boundingBoxToggleTitle = showBoundingBox ? "Hide bounding box" : "Show bounding box";
+    const turntableToggleTitle = autorotate ? "Turn off turntable" : "Turn on turntable";
+
+    return (
+      <div className={`viewer-toolbar-container${scrollMode ? " viewer-toolbar-scroll" : ""}`}>
+        <div
+          className="viewer-toolbar-scroll-left"
+          style={{ display: scrollBtnLeft ? "flex" : "none" }}
+          onClick={() => this.scrollX(-100)}
+        >
+          <ViewerIcon type="closePanel" style={{ fontSize: "12px", transform: "rotate(180deg)" }} />
```

Comment:
should these styles be set in a more systematic way?  Is the 12px something that gets reused in a lot of places, for example?

Reply:
I don't believe any other icon is quite this small, and given that `ViewerIcon` sets its own `style` property with a default font size (I'm not extremely happy with this pattern but can't think of anything that would make me happier), I think it makes sense to explicitly departure from the default size of 19px in props. Though in that case this prop could still be condensed down from a general style setter to a specific `size` prop and other styles set via plain CSS.